spec/classes/redis_sentinel_spec.rb

```diff
@@ -1,106 +1,103 @@
 require 'spec_helper'
 
-expected_noparams_content = <<EOF
-port 26379
-dir /tmp
-daemonize yes
-pidfile /var/run/redis/redis-sentinel.pid
-
-sentinel monitor mymaster 127.0.0.1 6379 2
-sentinel down-after-milliseconds mymaster 30000
-sentinel parallel-syncs mymaster 1
-sentinel failover-timeout mymaster 180000
-
-loglevel notice
-logfile /var/log/redis/redis.log
-EOF
-
-expected_params_content = <<EOF
-bind 1.2.3.4
-port 26379
-dir /tmp
-daemonize yes
-pidfile /var/run/redis/redis-sentinel.pid
-
-sentinel monitor cow 127.0.0.1 6379 2
-sentinel down-after-milliseconds cow 6000
-sentinel parallel-syncs cow 1
-sentinel failover-timeout cow 28000
-sentinel auth-pass cow password
-sentinel notification-script cow bar.sh
-sentinel client-reconfig-script cow foo.sh
-
-loglevel notice
-logfile /tmp/barn-sentinel.log
-EOF
-
-describe 'redis::sentinel', type: :class do
-  let(:facts) { debian_facts }
-
-  let :pre_condition do
-    [
-      'class { redis: }'
-    ]
-  end
-
-  describe 'without parameters' do
-    it { is_expected.to create_class('redis::sentinel') }
-
-    it {
-      is_expected.to contain_file('/etc/redis/redis-sentinel.conf.puppet').with(
-        'ensure'  => 'present',
-        'mode'    => '0644',
-        'owner'   => 'redis',
-        'content' => expected_noparams_content
-      )
-    }
-
-    it {
-      is_expected.to contain_service('redis-sentinel').with(
-        'ensure'     => 'running',
-        'enable'     => 'true',
-        'hasrestart' => 'true',
-        'hasstatus'  => 'true'
-      )
-    }
-  end
-
-  describe 'with custom parameters' do
-    let(:params) do
-      {
-        auth_pass: 'password',
-        sentinel_bind: '1.2.3.4',
-        master_name: 'cow',
-        down_after: 6000,
-        log_file: '/tmp/barn-sentinel.log',
-        failover_timeout: 28_000,
-        notification_script: 'bar.sh',
-        client_reconfig_script: 'foo.sh'
-      }
+describe 'redis::sentinel' do
+  on_supported_os.each do |os, facts|
+    context "on #{os}" do
+      let(:facts) { facts }
+      let(:config_file_orig) do
+        case facts[:osfamily]
+        when 'Archlinux'
+          '/etc/redis/redis-sentinel.conf.puppet'
+        when 'Debian'
+          '/etc/redis/redis-sentinel.conf.puppet'
+        when 'Suse'
+          '/etc/redis/redis-sentinel.conf.puppet'
+        when 'FreeBSD'
+          '/usr/local/etc/redis-sentinel.conf.puppet'
+        when 'RedHat'
+          '/etc/redis-sentinel.conf.puppet'
+        end
+      end
+
+      describe 'without parameters' do
+        let(:expected_content) do
+          <<~EOF
```

Comment:
This is Ruby 2.3 or 2.4 syntax so rubocop breaks on this. Anyone has any idea what the cleanest fix for this is? Should I use the classic `<<EOF` and not indent?